### ISSUE TITLE
`ATextarea`'s autofocus does not work reliably inside `AModal` with focus trap enabled

### ISSUE THREAD

**Issue by MikkCZ** [cisco-sbg-ui/magna-react#395]
**Describe the bug**
When `AModal` has the `withFocusTrap` enabled, the mechanisms tries to focus the first focusable element inside the modal content. This however competes with the `ATextarea` `autoFocus` prop and unfortunatelly, the focus trap wins.

**To Reproduce**
1. Have the following modal.
```js
export default function MyModal({isOpen}) {
  return (
    <AModal isOpen={isOpen} aria-labelledby="modalTitle" withFocusTrap>
      <APanel type="dialog">
        <APanelTitle id="modalTitle">TITLE</APanelTitle>
        <AButton tertiary>Focusable</AButton>
        <ATextarea autoFocus />
        <AButton primary>Also focusable</AButton>
      </APanel>
    </AModal>
  );
}
```
2. Open this modal.

**Actual behavior**
The textarea gets focused for a split of a second, but then the focus is stolen to the `AButton`, which is first focusable element in the modal.

**Expected behavior**
The element which is already focused in the modal (in our case the `ATextarea` via it's autofocus feature) stays focused.

**Screenshots**
<img width="501" alt="Screenshot 2023-05-19 at 9 11 56" src="https://github.com/cisco-sbg-ui/magna-react/assets/3941344/c2b8c84d-6f45-44a2-869f-f3d07f644b75">

**Comment by MikkCZ** [comment]
This is very similar to #396. However having a machanism to specify the initially focused element would not quite fully fix this.

In case the `ATextarea` is somewhere nested in a reusable component, it would become quite impractical to need to compine such mechanism with the `autoFocus` prop depending on whether the reusable component is rendered inside `AModal` or not.
